src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +392,402 @@ else if (!period.isEmpty()) {
+            }
+        }
+        return htmlString.toString();
+    }
+
+    /**
+     * Gets summary sentence for html inline type summary.
+     * Html type summary sentence are wrapped within paragraph tag.
+     *
+     * @param ast Javadoc root node.
+     * @return first sentence.
+     */
+    private static String getInlineHtmlSentence(DetailNode ast) {
+        final StringBuilder result = new StringBuilder(256);
+        final DetailNode inlineTagNode = getInlineTagNodeInsideHtml(ast);
+        result.append(getFirstSentenceFromInlineTagNode(
+                Objects.requireNonNull(inlineTagNode)));
+        return result.toString();
+    }
+
+    /**
+     * Finds and returns description node.
+     *
+     * @param ast Javadoc root node.
+     * @return Description DetailNode.
+     */
+    private static DetailNode getDescriptionNode(DetailNode ast) {
+        DetailNode descriptionNode = null;
+        for (DetailNode node : ast.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                for (DetailNode child : node.getChildren()) {
+                    if (child.getType() == JavadocTokenTypes.DESCRIPTION) {
+                        descriptionNode = child;
+                    }
+                }
+            }
+        }
+        return descriptionNode;
+    }
+
+    /**
+     * Tests if first sentence contains forbidden summary fragment.
+     *
+     * @param firstSentence String with first sentence.
+     * @return true, if first sentence contains forbidden summary fragment.
+     */
+    private boolean containsForbiddenFragment(String firstSentence) {
+        final String javadocText = JAVADOC_MULTILINE_TO_SINGLELINE_PATTERN
+                .matcher(firstSentence).replaceAll(" ").trim();
+        return forbiddenSummaryFragments.matcher(trimExcessWhitespaces(javadocText)).find();
+    }
+
+    /**
+     * Finds and returns first sentence from a DetailNode of inline form.
+     *
+     * @param node JavaDoc Inline node.
+     * @return first sentence.
+     */
+    private static String getFirstSentenceFromInlineTagNode(DetailNode node) {
+        final StringBuilder result = new StringBuilder(256);
+        for (DetailNode child : node.getChildren()) {
+            if (ALLOWED_TYPES.contains(child.getType())) {
+                final String text;
+                if (child.getChildren().length == 0) {
+                    text = child.getText();
+                }
+                else {
+                    text = getTextFromDescription(child);
+                }
+
+                result.append(text);
+            }
+        }
+        return result.toString();
+    }
+
+    /**
+     * Returns the text from an InlineTag Description.
+     *
+     * @param child InlineTag Node.
+     * @return text of InlineTag Description.
+     */
+    private static String getTextFromDescription(DetailNode child) {
+        final DetailNode[] child2 = child.getChildren();
```

Comment:
please do not use numbers in names ... fix whole method.

Reply:
Done. It was removed earlier but got into it again after I took back the last commit.